### ISSUE TITLE
Add a parameter corresponding to -canonList <file> option of snpEff ann

### ISSUE THREAD

**Issue by YeHW** [nf-core/modules#3822]
### Description of feature

`snpEff ann` command has an option: `-canonList <file>`, that can take in a tab-separated file containing gene and transcript.

> -canonList <file>            : Only use canonical transcripts, replace some transcripts using the \`gene_id      transcript_id\` entries in \<file\>.

Propose to add a parameter (e.g. `snpeff_canon_list`), whose value should be a string indicating the path of the canonList file - Allowing users to supply custom canonList file to be used with `snpEff ann` command (`SNPEFF_SNPEFF` process).

[Discussion on slack.](https://nfcore.slack.com/archives/CGFUX04HZ/p1693991106944439)
@maxulysse Thank you!

**Comment by maxulysse** [comment]
Hi @YeHW I transferred the issue in https://github.com/nf-core/modules

**Comment by YeHW** [comment]
> 

Yeah, it should be a tab-delimited file with two columns, corresponding to gene and transcript.
[canonList.txt](https://github.com/nf-core/modules/files/12554899/canonList.txt)

**Comment by YeHW** [comment]
> Hi @YeHW I transferred the issue in https://github.com/nf-core/modules

Sure, thanks!
